uat/custom-components/client-java-paho/src/main/java/com/aws/greengrass/testing/mqtt5/client/grpc/GRPCDiscoveryClient.java

```diff
@@ -122,6 +122,9 @@ public void onReceiveMqttMessage(int connectionId, MqttReceivedMessage message)
                                         .setPayload(ByteString.copyFrom(message.getPayload()))
                                         .setQos(MqttQoS.forNumber(message.getQos()))
                                         .setRetain(message.isRetain())
+                                        .setProperties(
+                                                Mqtt5Properties.newBuilder()
+                                                .putAllUserProperties(message.getUserProperties()).build())
```

Comment:
Do not call MessageBuilder.setProperties() in case when incoming message has no user properties at all.

Empty Hash not the same as  null.